### DISCUSSION TITLE
Create a red line with opaque white

### DISCUSSION THREAD

**Discussion by Raffinade** [ImageMagick/ImageMagick#4000] [Help]
Hi everybody, 
i want to create a red line in a picture with a white background (like a bookpage). The red line should be red to all areas that are white (like the background), all other parts of the line should not been filled (like grey, blue or any other pixels). 
I tried it with this command: 
`for i in *.TIF; do convert -fill red -fuzz 10% -stroke red -strokewidth 30 -draw 'line 10,0 20,3500' -opaque white "$i" "${i%.TIF}.tif" ; done`
But then the whole page is red except other colors that are not white. But I wanted that only in the red line with the strokewidth 30. 
Could someone give me a helpfull proposal?

Greetings 
Raffinade

**Comment by snibgo** [comment]
What version of IM, on what platform? I will assume v7 on bash.

I think you want to draw a line on an image. Where the image is close to white, you want the line to be red. Otherwise, the line should not be visible. Is that correct?

If so, you want to make pixels red where two conditions are both true: (a) the pixel is on a specified line (with anti-aliasing, I assume), and (b) the pixel was close to white.

[EDIT: I assume your input is opaque.]

The obvious solution uses a mask, with one of two methods:

1. Change all near-white pixels to red, and composite this over the input with a mask representing the line.

2. Draw a red line over the entire image, and composite this over the input with a mask representing the near-white pixels.

Method (2) is:
```
magick \
  in.png \
  \( -clone 0 \
     -fill Red \
     -draw "line 10,0 20,3500" \
     +write x1.png \
  \) \
  \( -clone 0 \
     -fuzz 10% -fill White -opaque White \
     -fuzz 0   -fill Black +opaque White \
     +write x2.png \
  \) \
  -compose Over -composite \
  out.png
```
If you have IM v6, use "convert" instead of "magick".

The writes to x1.png and x2.png are so you can see how this works. Both lines can be removed.

**Reply by Raffinade** [reply]
I use ImageMagick 6.9.10-23 on linux. The problem is, that booksites have on the edges after deskewing lines in grey or other colors. So i want to have a clear withe border. If i do this: 
`for i in *.tif; do convert -fill white -fuzz 20% -gravity west -region 50x0+0+0 -opaque "srgb(240,243,234)" +region  "$i" "${i%.tif}.TIF" ; done`
The fuzz is to hard, so if i have pictures at the edges of the site, you can see stripes in the picture. If i use a softer fuzz, the lines dont go away. 
I think that its better to draw a line exactly at the position the grey lines are and make fuzz and opaque, so that the stripes in pictures of the edge of the booksites are smaller. So i want to draw a stroke to do this. 
Before i do this, i want to make a red mask with stroke to see if the stroke is on the grey line. But if a draw a stroke, its only red and i cant see the grey lines through the red stroke. 
After that i want to delete the grey lines with the position of the stroke.

Greetings 

Raffinade

**Comment by snibgo** [comment]
I don't understand what you want. Please show example input images, and what results you want.
